src/components/AllComponents.js

```diff
@@ -11,13 +11,10 @@ import CardDetails from "./CardDetails";
 import Auth from "../util/auth";
```

Comment:
Avoided changing AllComponents as much as possible apart from doing a little cleanup on the useEffects, and then passing through the new form data to update the filteredCharacters